docs/execution-providers/EP-Context-Design.md

```diff
@@ -52,7 +52,7 @@ Atrribures:
 |ep.context_node_name_prefix|Used for context model generation only.<br/>Specify the EPContext node name (also the partition_name attribute, internal graph name) prefix to make it unique across nodes in case user glue multiple EPContext nodes in one model to avoid conflict.|
 |ep.context_model_external_initializers_file_name|This is for the case that some nodes partitioned on CPU EP, and those nodes has external initializers. When generating EP context model, the new generated model should NOT depend on old external data file used for source Onnx model.<br/>Use this config when dumping EP context model with an external initializers file. All initializers will be inside the external data file if specified, otherwise all inside generated Onnx file.<br/>It is not set by default, so all initializers will be inside the Onnx file.|
```

Comment:
>should NOT depend on old external data file used for source Onnx model. [](http://example.com/codeflow?start=211&length=71)

If we have ONNX model with external weights (e.g. 2GB) in data file. We can generate EP context which references weights data file instead of copying weights to new binary file? That would be significant disk footprint saving.

Reply:
Lets say some subgraphs partitioned to EP, and those subgraph depend on external data, and some subgraph fallback to CPU EP, and they also depend on external data.
For the parts partitioned to EP, they will be compiled into EP context binary file including the external data. nothing left in the generated _ctx.onnx model except EPContext nodes to reference the binary file.
Here, it's talk about the parts fallback to CPU EP, and they has external data dependency, we will embed the external data into the _ctx.onnx model by default. this is just for the external data used by those subgraphs.